updater.go

```diff
@@ -184,6 +182,80 @@ func (u *Updater) update(ctx Context, options UpdateOptions) (*Update, error) {
 	return update, nil
 }
 
+func (u *Updater) ApplyDownloaded(ctx Context) (bool, error) {
+	options := ctx.UpdateOptions()
+
+	// 1. check with the api server again for the latest update to be sure that a new update has not come out since our last call to CheckAndDownload
+	u.log.Infof("Attempting to apply previously downloaded update")
+	update, err := u.checkForUpdate(ctx, options)
+
+	if err != nil {
+		return false, findErr(err)
+	}
+
+	// Only report apply success/failure
+	applied, err := u.applyDownloaded(ctx, update, options)
+	if applied || err != nil {
+		report(ctx, err, update, options)
+	}
+	if err != nil && !applied {
+		report(ctx, err, update, options)
+	}
```

Comment:
second if condition seems covered by the first unless its meant to report twice?

Reply:
whooop! forgot to remove the first conditional thanks.